ci/jenkins/test.sh

```diff
@@ -408,12 +425,18 @@ function deliver_antrea_windows {
 
 function deliver_antrea {
     echo "====== Cleanup Antrea Installation ======"
-    clean_up_one_ns "antrea-ipam-test-11"
-    clean_up_one_ns "antrea-ipam-test-12"
-    clean_up_one_ns "antrea-ipam-test"
-    clean_up_one_ns "antrea-test"
+    if [[ $FLEXIBLE_IPAM == false ]]; then
+        clean_up_one_ns "antrea-ipam-test-11"
+        clean_up_one_ns "antrea-ipam-test-12"
+        clean_up_one_ns "antrea-ipam-test"
+        clean_up_one_ns "antrea-test"
+    fi
+    kubectl delete -f ${WORKDIR}/antrea-prometheus.yml || true
     kubectl delete daemonset antrea-agent -n kube-system || true
     kubectl delete -f ${WORKDIR}/antrea.yml || true
+    if [[ $FLEXIBLE_IPAM == true ]]; then
+        deliver_k8s
```

Comment:
Question - why we always need to redeploy K8s for flexible IPAM? If IP mode does not change, we need not to redeploy?

Please add some comments here about why we need to redeploy K8s.

Reply:
Ok. Let me know when you conclude.

Reply:
Added ip-mode check. It will redeploy only when ip-mode mismatch.